extensions/modules/file/src/test/xquery/modules/file/read-binary.xqm

```diff
@@ -19,19 +19,35 @@
  : License along with this library; if not, write to the Free Software
  : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  :)
-xquery version "3.0";
+xquery version "3.1";
 
-module namespace read-binary="http://exist-db.org/testsuite/modules/file/read-binary";
+module namespace readbinary="http://exist-db.org/testsuite/modules/file/read-binary";
+import module namespace helper="http://exist-db.org/xquery/test/util/helper" at "resource:util/helper.xqm";
+import module namespace fixtures="http://exist-db.org/xquery/test/util/fixtures" at "resource:util/fixtures.xqm";
 
 import module namespace test="http://exist-db.org/xquery/xqsuite" at "resource:org/exist/xquery/lib/xqsuite/xqsuite.xql";
 import module namespace file="http://exist-db.org/xquery/file";
+import module namespace util="http://exist-db.org/xquery/util";
 
+declare variable $readbinary:suite := "read-binary";
 
 declare
-    %test:pending("TODO need to mechanism to setup a temporary file to work with")
-    %test:assertEquals("<root>bla</root>")
-function read-binary:without-serialization() {
-  let $binData := file:read-binary("VERSION.txt")
-  return
-    element { "root" } { "bla" }
+    %test:setUp
+function readbinary:setup() as empty-sequence() {
+};
+
+declare
+    %test:tearDown
+function readbinary:tear-down() as empty-sequence() {
+    helper:clear-suite-fs($readbinary:suite)
+};
+
+declare
+    %test:assertEquals("SERVER_SECRET=123!")
+function readbinary:without-serialization() {
+    helper:get-test-directory($readbinary:suite)
+    => helper:setup-fs-extra()
```

Comment:
I would suggest removing the arrow operator here, it will make Java debugging of this code almost impossible to follow - test code should be as simple as possible rather than clever.